evalml/automl/automl_search.py

```diff
@@ -494,12 +498,26 @@ def search(self, show_iteration_plot=True):
                 logger.info('AutoML Algorithm out of recommendations, ending')
                 break
             try:
-                new_pipeline_ids = self._engine.evaluate_batch(current_batch_pipelines)
+                computations = []
+                new_pipeline_ids = []
+                log_title(logger, f"Evaluating Batch Number {self._get_batch_number()}")
+                for pipeline in current_batch_pipelines:
+                    self._pre_evaluation_callback(pipeline)
+                    computation = self._engine.submit_evaluation_job(self.automl_data, pipeline, self.X_train, self.y_train)
+                    computations.append(computation)
+                while self._should_continue() and len(computations) > 0:
```

Comment:
what is the behavior of this when using the dask engine and all pipelines in a batch are currently computing? does it just iterate as fast as it can over and over trying to find whatever computation has completed? 

I wonder if this code can be simplified using a method like: https://docs.dask.org/en/latest/futures.html#distributed.as_completed

Reply:
You're right about the current behavior! I thought about using/writing our own `as_completed` but the one tricky thing is that we want to stop the search as soon as one of our stopping criteria is met but `as_completed` will only pull off futures if they're done. I think this will make it hard to respect time based stopping rules. 

For example, let's say the user has a 2 hour stop limit and fires off four jobs. The first three finish in 90 minutes but the fourth one won't finish until 6 hours.

If we do this:

```python
for computation in as_completed(computations):
    if not self._should_continue():
         break
```

We actually won't reach the break until hour 6 because that's when the final future finished.

I think we can definitely explore ways to consolidate/simplify the logic for pulling futures off of our queue but I'd rather leave that improvement for a separate PR.

Reply:
i wonder if we need some sort of sleep call in the loop then. if you have a bunch of pipelines training for a while, i dont think you want this loop iterating as fast as it can.